apsuite/commisslib/measure_tune_bpm.py

```diff
@@ -0,0 +1,287 @@
+        data['init_delay'] = self.delay0
+        self.data = data
+
+    def load_orbit(self, data=None, orbx=None, orby=None):
+        """Load orbit data into the object. You can pass the
+        intire data dictionary or just the orbits. If data argument
+        is provided, orbx and orby become optional"""
+
+        if not hasattr(self, 'data'):
+            self.data = dict()
+
+        if data is not None:
+            self.data = data
+        if orbx is not None:
+            self.data['orbx'] = orbx
+        if orby is not None:
+            self.data['orby'] = orby
+
+    def dft(self, bpm_indices=None):
+        """Apply a dft at bpms.
+
+        Args:
+        - bpm_indices (int, list or np.array): BPM indices whose dft will
+        be applied. Default is return the dft of all bpms.
+
+        Returns:
+         - spectrumx, spectrumy, freqs: The first two are spectra np.arrays
+         of dimension #freqs x #bpm_indices, and freqs is a np.array with
+         the frequency domain values.
+        """
+        if bpm_indices is not None:
+            orbx = self.data['orbx'][:, bpm_indices]
+            orby = self.data['orby'][:, bpm_indices]
+        else:
+            orbx = self.data['orbx']
+            orby = self.data['orby']
+
+        x_beta = orbx - orbx.mean(axis=0)
+        y_beta = orby - orby.mean(axis=0)
+
+        N = x_beta.shape[0]
+        freqs = rfftfreq(N)
+
+        if isinstance(bpm_indices, int):
+            spectrumx = _np.abs(rfft(x_beta))
+            spectrumy = _np.abs(rfft(y_beta))
+        else:
+            spectrumx = _np.abs(rfftn(x_beta, axes=[0]))
+            spectrumy = _np.abs(rfftn(y_beta, axes=[0]))
+
+        return spectrumx, spectrumy, freqs
+
+    def naff_tunes(self, dn=None, window_param=1, bpm_indices=None):
+        """Computes the tune evolution from the BPMs matrix with a moving
+            window of length dn.
+           If dn is not passed, the tunes are computed using all points."""
+
+        if bpm_indices is not None:
+            x = self.data['orbx'][:, bpm_indices]
+            y = self.data['orby'][:, bpm_indices]
+        else:
+            x = self.data['orbx']
+            y = self.data['orby']
+        N = x.shape[0]
+
+        if dn is None:
+            return self.tune_by_naff(x, y)
+        else:
+            tune1_list = []
+            tune2_list = []
+            slices = _np.arange(0, N, dn)
+            for idx in range(len(slices)-1):
+                idx1, idx2 = slices[idx], slices[idx+1]
+                sub_x = x[idx1:idx2, :]
+                sub_y = y[idx1:idx2, :]
+                tune1, tune2 = self.tune_by_naff(sub_x, sub_y, window_param=1,
+                                                 decimal_only=True)
+                tune1_list.append(tune1)
+                tune2_list.append(tune2)
+
+        return _np.array(tune1_list), _np.array(tune2_list)
+
+    def spectrogram(self, dn=None, overlap=True, bpm_indices=None):
```

Comment:
maybe, instead of a boolean variable, the argument `overlap` could be an integer where the user can choose how much overlap is used. I was thinking on an interface similar to the [welch method](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.welch.html) of scipy.

Reply:
I discovered that scipy has a function to make [spectrograms](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.spectrogram.html?highlight=spectrogram#scipy.signal.spectrogram) with the overlap option that you cite and others useful arguments. 

In the next commit, I will rewrite this class method to incorporate the scipy.signal.spectrogram function.

Reply:
Done! Thanks for the suggestions.